src/Nethermind/Nethermind.Optimism/CL/P2P/OptimismCLP2P.cs

```diff
@@ -132,6 +175,64 @@ private async Task MainLoop(CancellationToken token)
         }
     }
 
+    private async Task<ExecutionPayloadV3?> TryRequestPayload
+        (ISession? session, ulong payloadNumber, Hash256 expectedHash, CancellationToken token)
+    {
+        try
+        {
+            CancellationTokenSource cancellationTokenSource = CancellationTokenSource.CreateLinkedTokenSource(token);
```

Comment:
Ensure `CancellationTokenSource` gets disposed:

```suggestion
            using CancellationTokenSource cancellationTokenSource = CancellationTokenSource.CreateLinkedTokenSource(token);
```